src/analyses/baseInvariant.ml

```diff
@@ -581,8 +581,8 @@ struct
       else
         match exp, c_typed with
         | UnOp (LNot, e, _), Int c ->
-          (match Cilfacade.typeOf e with
-           | TInt  _ | TPtr _ ->
+          (match Cil.unrollType (Cilfacade.typeOf e) with
+           | TInt  _ | TPtr _ -> (* TODO: TEnum as well? *)
```

Comment:
Yeah, `TEnum` is `TInt` for all intents and purposes, so we should add it here.

Reply:
I might also look through other uses of `TInt` to see if `TEnum` is handled consistently as well, if we're already in the process of doing these things.

Reply:
I've now done that in most places.